Add `sails-js` (#167)
gear-foundation/vara-wiki: docs/sails-js/react-hooks.md

    library: Program,
    id: '0x...',
  });

  const { data } = useProgramQuery({
    program,
    serviceName: 'service',
    functionName: 'query',
    args: ['arg', 'anotherArg'],
    watch: false, // If true, initializes a subscription to the program's state changes in the Gear MessagesDispatched event. Can increase network traffic.
  });

  return <div>{JSON.stringify(data)}</div>;
}
```

### `useProgramEvent`

The `useProgramEvent` hook initializes a subscription to a specified program event.

**Example:**

The `useProgramEvent` hook below sets up a listener on the `event` function of the specified `service` and executes the `onData` callback with the new data whenever the event is triggered.

```jsx
import { useProgram, useProgramEvent } from '@gear-js/react-hooks';
import { Routing } from './pages';
import { Program } from './lib';

function App() {
  const { data: program } = useProgram({
    library: Program,
    id: '0x...',
  });

  useProgramEvent({
    program,
    serviceName: 'service',
    functionName: 'event',
    onData: (data) => console.log(data),
  });

  return (
    <main>
      <Routing />
    </main>
  );
}

export { App };
```

## Prepare Hooks

> When performing long-running asynchronous work in response to a user interaction (e.g., using "transactional" hooks such as [`useSendProgramTransaction`](#usesendprogramtransaction)), the user only receives feedback on the transaction status after invoking the click handler. Prepare hooks can be used to eagerly obtain values before user interaction or to perform upfront validation before sending a transaction.

Reviewer: ```suggestion
When performing long-running asynchronous work in response to a user transaction using [`useSendProgramTransaction`](#usesendprogramtransaction)) hook, the user only receives feedback on the transaction status after transaction is complete. Prepare hooks can be used to eagerly obtain values before user interaction or to perform upfront validation before sending a transaction.
```

I guess, we can be specific here. As of right now there's only one purpose.
